File: map_backend/src/models/message_model.js
import mongoose from "mongoose";


const messageSchema= mongoose.Schema({
    senderId: {
        type: mongoose.Schema.ObjectId,
        ref: 'User',
        required: true
    },
    content: {
        type: String,
        required: true,
    },
    conversationId: {
        type: mongoose.Schema.ObjectId,
        ref: 'Conversation',
        required: true
    },
    timestamps: {
        type: Date,
        default: Date.now()
    }
}, {
    timestamps: true
});

const Message=mongoose.model('Message', messageSchema);

export default Message;

File: new_project/lib/models/message_model.dart
class MessageModel {
  final String? senderId;
  final String? content;
  final String? conversationId;
  final String? timestamps;

  MessageModel(
      {this.content, this.conversationId, this.senderId, this.timestamps});

  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
        content: json['content'] ?? '',
        conversationId: json['conversationId'] ?? '',
        senderId: json['senderId'] ?? '',
        timestamps: json['timestamps'] ?? '');
  }
}
